internal/pkg/comm/client.go

```diff
@@ -71,7 +71,9 @@ func (client *GRPCClient) parseSecureOptions(opts SecureOptions) error {
 
 	client.tlsConfig = &tls.Config{
 		VerifyPeerCertificate: opts.VerifyCertificate,
-		MinVersion:            tls.VersionTLS12} // TLS 1.2 only
+		MinVersion:            tls.VersionTLS12,
+		MaxVersion:            tls.VersionTLS13,
```

Comment:
NoOps here , but you could have 0 also as MaxVersion value. To me , it seems logical to have 0 consideration the situation when TLS moves to say , 1.4 , we do not have to change this field.

Reply:
> One overall comment : The list of default Cipher Suite contains one Cipher more than what TLS 1.3 supports. Not knowing , how TLS 1.3 loads Cipher
> 
> 1. Does it load from its engine or
> 2. Honor defaultConfig setup in the code.
> 
> If it is latter , then it might throw error saying Cipher XXXXX not supported.
> 
> To be on safer side , we need to cipher list in defaultCipherSuite matching TLS 1.3 cipher IMO.

Yea, I'm working on that one now, I didn't run the tests locally, so I didn't get the failed cipher until it finished testing in CI